backend/api/Controllers/MissionController.cs

```diff
@@ -0,0 +1,84 @@
+﻿using System.Text.Json;
+using Api.Models;
+using Api.Services;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Api.Controllers;
+
+[ApiController]
+[Route("missions")]
+public class MissionController : ControllerBase
+{
+    private readonly ILogger<MissionController> _logger;
+
+    private readonly EchoService _echoService;
+
+    public MissionController(ILogger<MissionController> logger, EchoService echoService)
+    {
+        _logger = logger;
+        _echoService = echoService;
+    }
+
+    /// <summary>
+    /// List all available missions on the asset
+    /// </summary>
+    /// <remarks>
+    /// Overview
+    ///
+    /// List all available missions on the asset in the Echo mission planner
+    /// </remarks>
+    [HttpGet]
+    [ProducesResponseType(typeof(List<Mission>), StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+    [ProducesResponseType(StatusCodes.Status403Forbidden)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    public async Task<ActionResult<IList<Mission>>> GetMissions()
+    {
+        try
+        {
+            var missions = await _echoService.GetMissions();
+
+            return Ok(missions);
+        }
+        catch (Exception e) when (e is MissionNotFoundException || e is JsonException)
+        {
+            _logger.LogError(e, "Missions not found");
+
+            return NotFound();
```

Comment:
Do we want to add a comment here? Something like:

```
NotFound("$Missions not found: {error}", e)
```

Is it necessary with both logging and returning a not found? I think we should do this similarly for all controllers to keep a readable log.

Reply:
What is added to the `NotFound` statement is not logged to the application log, it's sent to the user. So the log statement is what we wish to log in our application while the `NotFound` statement is what we wish to tell the user. 

Do we want to provide the full stack trace to the user? I would think it's more useful to log the stack trace in the application and provide a not found to the user.

Reply:
As for `NotFound()` vs ``NotFound("Missions not found") I think they give about the same level of information if you're not adding the error message. I would rather have an empty `NotFound()` maybe than have to maintain specific strings throughout the application for different types of `NotFound()`?

Reply:
I agree with your points @aestene . I would just point out that if its not clear "what" was NotFound, then it might be worth adding it as a parameter, for example when we call robots/{id}/missions/{id}, We should clarify whether it was the robot or the mission that was not found

Reply:
Sounds good, @aestene and @GodVenn. I think we should make sure that the logging is similar across the controllers, but that can be handled in #199.